File: problems/problems_LCR_008/solution.py
from collections import deque
from math import inf

import solution
from typing import *


class Solution(solution.Solution):
    def solve(self, test_input=None):
        return self.minSubArrayLen(*test_input)

    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        q = deque()
        cur_sum = 0
        ans = inf
        for num in nums:
            q.append(num)
            cur_sum += num
            while cur_sum >= target:
                ans = min(ans, len(q))
                cur_sum -= q.popleft()
        return ans if ans != inf else 0

File: problems/problems_LCR_008/Solution.java
package problems.problems_LCR_008;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public int minSubArrayLen(int target, int[] nums) {
        Deque<Integer> queue = new ArrayDeque<>();
        int sum = 0;
        int n = nums.length;
        int ans = n + 1;
        for (int num: nums) {
            queue.addLast(num);
            sum += num;
            while (sum >= target) {
                ans = Math.min(ans, queue.size());
                Integer first = queue.pollFirst();
                sum -= first == null ? 0 : first;
            }
        }
        return ans == n + 1 ? 0 : ans;
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        int target = Integer.parseInt(inputJsonValues[0]);
		int[] nums = jsonArrayToIntArray(inputJsonValues[1]);
        return JSON.toJSON(minSubArrayLen(target, nums));
    }
}
